sympy/stats/rv.py

```diff
@@ -1221,7 +1247,7 @@ def sampling_P(condition, given_condition=None, numsamples=1,
     count_true = 0
     count_false = 0
 
-    samples = sample_iter(condition, given_condition,
+    samples = sample_iter(condition, given_condition, size=size, library=library,
```

Comment:
I see you have added `size` argument to some functions.
But does the array shape make sense for mean, variance if numsamples is already used?

Reply:
The actual problem, I see is that is `size` and `numsamples` mean the same, as before, `numsamples` was not provided in public API of `sample`, and so, `size` was introduced. So, we need to take a decision to use either `size` or `numsamples`. We need to raise Deprecation warning in either of the case.

Reply:
IMO, we should remove `size`, and use `numsamples` as it was been there for a long period of time.

Reply:
I observed many failing things:
```
>>> from sympy.stats import *
>>> X=Normal('X',1,2)
>>> sample(X)     #works fine
-0.395115576398039

>>> sample(X, size=3)     # works fine
[3.04853254256568, -5.15781049701721, -0.916547668739276]
>>> Y=Normal('Y',3,4)
>>> sample(X+Y, size=3)   # failed : expected 3
2.94035736693013

>>> sample(X+Y, numsamples=3)   # failed : expected 3
5.91353130029414

>>> # adding numsamples as argument in sample function
>>> sample(X+Y, numsamples=3)   # works fine
[2.60480735236754, -2.59005952144665, 1.87621810047909]

>>> sample(X+Y, size=3)   # failed : expected 3
[1.22518985609606]

>>> sample(X+Y, numsamples=3, library='scipy') # fails : expected to produce array output
[1.18343224436589, 5.09387064093793, 5.55350069910508]

>>> sample(X, numsamples=3, library='scipy') # fails : expected to produce array output
[3.9671541494809617, -2.810799332922929, -1.0880356350959333]

>>> sample(X, size=3, library='scipy')  # failed : expected array instead of list
[array([ 2.02042956, -0.71548653,  2.01273328])]
```

Reply:
Looks quite inconsistent. What is the difference between, `numsamples` and `size`?

Reply:
What is used in public API, `size` or `numsamples`?

Reply:
Actually, `numsamples` has previously used for signifying total number of samples, but the user cannot provide it while calling` sample()`(as shown 4th example which doesn't affect currenlty as is overwritten), therefore `size` was introduced for the same use, but now I see that `size` and `numsamples` both mean the same.

Reply:
So, I suggest removing `size` and use `numsamples` as it will maintain unifromity across all other samplings in `rv.py` and avoid confusion

Reply:
So, now `size` is what public API has, so `numsamples` should be removed/renamed to `size`.

Reply:
Some functions like `sampling_P`, `sampling_density`, etc. have `numspaces`

Reply:
https://github.com/sympy/sympy/blob/63ab534ebc9bb8fa47ca80aea5a1b0eca743fb9e/sympy/stats/rv.py#L1020-L1040
In `sample` we have `size` as public API, while returning `numsaples` is by default kept as 1.

The following samplings have `size` and `numsamples` both as the argument.
https://github.com/sympy/sympy/blob/63ab534ebc9bb8fa47ca80aea5a1b0eca743fb9e/sympy/stats/rv.py#L1043

https://github.com/sympy/sympy/blob/63ab534ebc9bb8fa47ca80aea5a1b0eca743fb9e/sympy/stats/rv.py#L1130

https://github.com/sympy/sympy/blob/63ab534ebc9bb8fa47ca80aea5a1b0eca743fb9e/sympy/stats/rv.py#L1177


The following samplings have only `numsamples` as argument.
https://github.com/sympy/sympy/blob/63ab534ebc9bb8fa47ca80aea5a1b0eca743fb9e/sympy/stats/rv.py#L1207

https://github.com/sympy/sympy/blob/63ab534ebc9bb8fa47ca80aea5a1b0eca743fb9e/sympy/stats/rv.py#L1243

https://github.com/sympy/sympy/blob/63ab534ebc9bb8fa47ca80aea5a1b0eca743fb9e/sympy/stats/rv.py#L1265

Reply:
@czgdp1807 Please share your thoughts on removing which of this argument, to make it uniform throughout all these functions.

Reply:
What happens if we just let the sampling happen by using default values of `numsamples` and `size` in the above functions? 
The removal should be done in a different PR. Can you open an issue for it as it is concerned with the changes in public API.